database_cleanup/models/purge_tables.py

```diff
@@ -64,75 +44,62 @@ def purge(self, cr, uid, ids, context=None):
                 """ % line.name)
 
-            for constraint in cr.fetchall():
+            for constraint in self.env.cr.fetchall():
                 if constraint[3] in tables:
                     self.logger.info(
                         'Dropping constraint %s on table %s (to be dropped)',
                         constraint[0], constraint[3])
-                    cr.execute(
+                    self.env.cr.execute(
                         "ALTER TABLE %s DROP CONSTRAINT %s" % (
                             constraint[3], constraint[0]))
 
             self.logger.info(
                 'Dropping table %s', line.name)
-            cr.execute("DROP TABLE \"%s\"" % (line.name,))
+            self.env.cr.execute("DROP TABLE \"%s\"" % (line.name,))
             line.write({'purged': True})
-            cr.commit()
+            self.env.cr.commit()
         return True
 
 
-class CleanupPurgeWizardTable(orm.TransientModel):
+class CleanupPurgeWizardTable(models.TransientModel):
     _inherit = 'cleanup.purge.wizard'
     _name = 'cleanup.purge.wizard.table'
+    _description = 'Purge tables'
 
-    def default_get(self, cr, uid, fields, context=None):
-        res = super(CleanupPurgeWizardTable, self).default_get(
-            cr, uid, fields, context=context)
-        if 'name' in fields:
-            res['name'] = _('Purge tables')
-        return res
-
-    def find(self, cr, uid, context=None):
+    @api.model
+    def find(self):
         """
         Search for tables that cannot be instantiated.
         Ignore views for now.
         """
-        model_ids = self.pool['ir.model'].search(cr, uid, [], context=context)
         # Start out with known tables with no model
         known_tables = ['wkf_witm_trans']
-        for model in self.pool['ir.model'].browse(
-                cr, uid, model_ids, context=context):
-
-            model_pool = self.pool.get(model.model)
-            if not model_pool:
+        for model in self.env['ir.model'].search([]):
+            if model.model not in self.env:
                 continue
+            model_pool = self.env[model.model]
             known_tables.append(model_pool._table)
             known_tables += [
                 column._sql_names(model_pool)[0]
                 for column in model_pool._columns.values()
                 if (column._type == 'many2many' and
                     hasattr(column, '_rel'))  # unstored function fields of
                                               # type m2m don't have _rel
-                ]
+            ]
 
         # Cannot pass table names as a psycopg argument
         known_tables_repr = ",".join(
             [("'%s'" % table) for table in known_tables])
-        cr.execute(
+        self.env.cr.execute(
             """
             SELECT table_name FROM information_schema.tables
             WHERE table_schema = 'public' AND table_type = 'BASE TABLE'
             AND table_name NOT IN (%s)""" % known_tables_repr)
```

Comment:
Is posible avoid sql-injection here?
`s/% known_tables_repr/, (known_tables_repr,)/g`